test/unit/mock/readable-storage-logic.spec.ts

```diff
@@ -0,0 +1,175 @@
+    });
+
+    it('should be able to get a int56', async () => {
+      const value = 1;
+      await mock.setVariable('_int56', value);
+      expect(await mock.getInt56()).to.equal(value);
+
+      const getValue = await mock.getVariable('_int56');
+      const expectedValue = await mock.getInt56();
+      expect(getValue.toString()).to.equal(expectedValue.toString());
+    });
+
+    // it('should be able to get a int256', async () => {
+    //   const value = utils.parseUnits('-1');
+    //   await mock.setVariable('_int256', value);
+
+    //   const getValue = await mock.getVariable('_int256');
+    //   const expectedValue = await mock.getInt256();
+    //   console.log("getValue: ", getValue);
+    //   expect(getValue).to.equal(expectedValue.toString());
+    // });
+
+    it('should be able to get an address', async () => {
+      await mock.setVariable('_address', ADDRESS_EXAMPLE);
+      expect(await mock.getAddress()).to.equal(ADDRESS_EXAMPLE);
+
+      const getValue = await mock.getVariable('_address');
+      const expectedValue = await mock.getAddress();
+      expect(getValue).to.equal(expectedValue);
+    });
+
+    it('should be able to get a boolean', async () => {
+      await mock.setVariable('_bool', true);
+      expect(await mock.getBool()).to.equal(true);
+
+      const getValue = await mock.getVariable('_bool');
+      const expectedValue = await mock.getBool();
+      expect(getValue).to.equal(expectedValue);
+    });
+
+    it('should be able to get a small value < 31 bytes', async () => {
+      const value = BYTES_EXAMPLE.slice(0, 10);
+      await mock.setVariable('_bytes', value);
+      expect(await mock.getBytes()).to.equal(value);
+
+      const getValue = await mock.getVariable('_bytes');
+      const expectedValue = await mock.getBytes();
+      expect(getValue).to.equal(expectedValue);
+    });
+
+    it('should be able to get a large value > 31 bytes', async () => {
+      await mock.setVariable('_bytes', BYTES_EXAMPLE);
+      expect(await mock.getBytes()).to.equal(BYTES_EXAMPLE);
+
+      const getValue = await mock.getVariable('_bytes');
+      const expectedValue = await mock.getBytes();
+      expect(getValue).to.equal(expectedValue);
+    });
+
+    it('should be able to get bytes32', async () => {
+      await mock.setVariable('_bytes32', BYTES32_EXAMPLE);
+      expect(await mock.getBytes32()).to.equal(BYTES32_EXAMPLE);
+
+      const getValue = await mock.getVariable('_bytes32');
+      const expectedValue = await mock.getBytes32();
+      expect(getValue).to.equal(expectedValue);
+    });
+
+    it('should be able to set a simple struct', async () => {
+      const struct = {
+        valueA: BigNumber.from(1234),
+        valueB: true,
+      };
+      await mock.setVariable('_simpleStruct', struct);
+
+      expect(convertStructToPojo(await mock.getSimpleStruct())).to.deep.equal(struct);
+
+      const getValue = await mock.getVariable('_simpleStruct');
```

Comment:
just asking here, what's `getValue` type? does `getValue.valueA` exists?

Reply:
In the case of struct `getValue` returns an object more specific here it returns: 
```
{
  valueA: BigNumber { _hex: '0x04d2', _isBigNumber: true },
  valueB: true
}
```

and `getValue.valueA` will give tou `BigNumber { _hex: '0x04d2', _isBigNumber: true }`